openssl/params.go

```diff
@@ -99,3 +101,14 @@ func (pb *paramsBuilder) addBigNumber(key *C.char, v []byte) error {
 	pb.add(key, C.GO_OSSL_PARAM_UNSIGNED_INTEGER, unsafe.Pointer(cbytes), C.size_t(len(v)))
 	return nil
 }
+
+// addBN adds a parameter of type OSSL_PARAM_UNSIGNED_INTEGER.
+func (pb *paramsBuilder) addBN(key *C.char, v C.GO_BIGNUM_PTR) error {
+	size := (C.go_openssl_BN_num_bits(v) + 7) / 8
+	x := C.malloc(C.size_t(size))
+	if C.go_openssl_BN_bn2nativepad(v, (*C.uchar)(x), size) == 0 {
+		return newOpenSSLError("BN_bn2nativepad")
```

Comment:
Would this be a memory leak in the error case?

Reply:
Good catch.